blogposts/2021/how-to-not-break-a-search-engine-or-what-i-learned-about-unglamorous-engineering.md

```diff
@@ -247,24 +252,22 @@ behavior is intentional. But a surprised or mildly inconvenienced user is a far
 cry from releasing a bug that takes down a company's instance. Severe bugs
 cascade into out-of-band releases for our distribution team and upgrades for
 customers. They also tend to have a latent effect on engineer productivity (in
-this case, mine) when bugs later impose context-switches to fix things.
-Conceptually big costs that I wanted to avoid. All of these considerations,
+this case, mine) when bugs later impose context-switches to fix things—conceptually big costs that I wanted to avoid. All of these considerations,
 code changes, and heaps of testing happened in the pursuit of an unglamorous
 outcome while two, maybe three, engineers reviewed the code to see it play out.
 I know this isn't an isolated thing. I get faint hints of other engineers at
 Sourcegraph doing momentous but unglamorous things that most of the
 organization is blissfully unaware of. And the Twitterverse suggests there's
 more of it happening in software all around us:
 
-<p style="padding: 0px 40px 0px 40px; text-align: justify;">“<i>
-A huge problem in software companies is that large new features get praise, promotions, accolades... while migrating off a legacy system, increasing performance ... or reducing error rates ... is often only recognized by peers...</i>” <a href="https://twitter.com/danmayer/status/139556425230854144"><b>↗</b></a></p>
+<blockquote class="twitter-tweet tw-align-center"><p lang="en" dir="ltr">A huge problem in software companies is that large new features get praise, promotions, accolades... while migrating off a legacy system, increasing performance 2,4,10X, or reducing error rates, pages, or alerts by X% is often only recognized by peers and not leadership.</p>&mdash; Dan Mayer (@danmayer) <a href="https://twitter.com/danmayer/status/1395564252308541440?ref_src=twsrc%5Etfw">May 21, 2021</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
 
 I empathize with the engineers who don't have an audience for their unglamorous
 work, who want to say "I did A Thing, there's nothing to see, but more people
-should care. Let me tell you about it!".  I like my portion of showpiece
+should care. Let me tell you about it!"  I like my portion of showpiece
 engineering, don't get me wrong. But shouldn't doing the necessary, unglamorous
 work be a marketable skill as well? Where's the signage that reads "Unglamorous
-engineers wanted. Will pay handsomely."? I hope you're encouraged to share what
+engineers wanted. Will pay handsomely"? I hope you're encouraged to share what
```

Comment:
Look. Punctuation around quotation marks is awkward and I don't think anyone has solved it yet, but I'm generally in favour of fewer punctuation marks